tuna/rocmlir/rocmlir_tables.py

```diff
@@ -0,0 +1,296 @@
+      'kernel_repeats': None,
+      'id': None,
+      'valid': None
+  }
+
+  def config_string(self):
+    """Return config as a flag/value string suitable for tuningRunner.py."""
+    string = "conv "  # +++pf:  of course generalise for gemm
+    for field, value in self.to_dict().items():
+      flag = self.options[field]
+      if flag:
+        string += f"{flag} {value} "
+    string += "-m conv"
+    return string
+
+
+class ConvolutionResults(BASE):  # pylint: disable=too-many-instance-attributes
+  """Collects the results of convolution tuning.
+  """
+  __tablename__ = "rocmlir_conv_results"
+  __table_args__ = (UniqueConstraint("config", "session", name="uq_idx"),)
+
+  @orm.reconstructor
+  def __init__(self, **kwargs):
+    if 'logger' in kwargs:
+      self.logger = kwargs['logger']
+    else:
+      self.logger = setup_logger('results')
+
+  @declared_attr
+  def session(self):
+    """session column"""
+    return Column(Integer, ForeignKey("session_rocmlir.id"), nullable=False)
+
+  config = Column(Integer, ForeignKey("rocmlir_conv_config.id"), nullable=False)
+  config_str = Column(Text, nullable=False)
+
+  perf_config = Column(Text, nullable=False)
+  kernel_tflops = Column(Float, nullable=False)
+
+  def get_query(self, sess, result_obj, session_id):
+    """Construct a Db query for the find object
+    """
+    query = sess.query(result_obj).filter(result_obj.session == session_id,
+                                          result_obj.config == self.config)
+    self.logger.info("result query %s-%s", session_id, self.config)
+    return query
+
+  # +++pf:  rewrite me for tuningRunner.py output!
+  def parse(self, lines):
+    """parse logger output line for result data """
+    line = lines.splitlines()[-1]
+    print(f"line being parsed is '{line}'", file=sys.stderr)
+    return line.split('\t')
+
+
+#pylint: disable=too-few-public-methods
+class RocMLIRDBTables(DBTablesInterface):
+  """Represents db tables for rocMLIR lib"""
+
+  def __init__(self, **kwargs):
+    """Constructor"""
+    super().__init__(**kwargs)
+    allowed_keys = set(['config_type', 'session_id'])
+
+    self.config_type = None
+    self.job_table = None
+    self.session_table = SessionRocMLIR
+    self.config_table = None
+    self.results = None
+
+    self.__dict__.update(
+        (key, value) for key, value in kwargs.items() if key in allowed_keys)
+    self.set_tables()
+
+  def set_tables(self, sess_class=SessionRocMLIR):
+    """Set appropriate tables based on requirements"""
+    super().set_tables(sess_class)
+    # +++pf:  branch on self.config_type when we have GEMM, too.
+    self.job_table = ConvolutionJob
+    self.config_table = ConvolutionConfig
+    self.results = ConvolutionResults
+
+
+def get_tables() -> List[BASE]:
+  """Returns a list of all Example lib DB tables"""
+  tables: List[BASE] = []
+  with DbSession() as session:
+    engine = session.bind
+    connect = session.connection()
+
+    def append_if_not_exists(table):
```

Comment:
There's a name clash with the Machine table, and enough of that class is in the base code that I didn't think I should change the name.  I also got tired of bumping into existing tables when testing.  Thus, append-if-not-exists to avoid double-creation.